src/software/gui/robot_diagnostics/threaded_robot_diagnostics_gui.cpp

```diff
@@ -8,13 +8,25 @@ ThreadedRobotDiagnosticsGUI::ThreadedRobotDiagnosticsGUI(int argc, char** argv)
       termination_promise_ptr(std::make_shared<std::promise<void>>()),
       sensor_msg_buffer(
           std::make_shared<ThreadSafeBuffer<SensorProto>>(sensor_msg_buffer_size)),
-      primitive_buffer(
-          std::make_shared<ThreadSafeBuffer<std::unique_ptr<TbotsProto::PrimitiveSet>>>(
-              primitive_buffer_size)),
+      primitive_buffer(std::make_shared<ThreadSafeBuffer<TbotsProto::PrimitiveSet>>(
+          primitive_buffer_size)),
       application_shutting_down(false)
 {
     run_robot_diagnostics_thread = std::thread(
         &ThreadedRobotDiagnosticsGUI::createAndRunRobotDiagnosticsGUI, this, argc, argv);
+    run_send_primitives_thread = std::thread([this]() {
+        while (true)
+        {
+            auto primitive = primitive_buffer->popMostRecentlyAddedValue();
+            if (primitive)
+            {
+                this->sendValueToObservers(std::move(primitive.value()));
+            }
+
+            std::this_thread::sleep_for(
```

Comment:
non-blocking nit: This could also be done with a `QTimer` to trigger a function call on an interval. Then you don't need a thread.

Reply:
dont think this is possible becuase the thread robot diagnostics gui isn't a q_object

Reply:
Oops you're right.